cmd/loki/main.go

```diff
@@ -43,11 +43,9 @@ func main() {
 
 	util.InitLogger(&cfg.Server)
 
-	if configFile != "" {
-		if err := helpers.LoadConfig(configFile, &cfg); err != nil {
-			level.Error(util.Logger).Log("msg", "error loading config", "filename", configFile, "err", err)
-			os.Exit(1)
-		}
+	if err := helpers.LoadConfig(configFile, &cfg); err != nil {
```

Comment:
Isn't this a breaking change, given the config file is not strictly required as of now and config can be specific via individual cli flags?

Reply:
Okay, you're right, I didn't realize this.

I will add the check back, but this requires your #721 to be merged, otherwise users will experience the `SEGFAULT`. Thanks!

Reply:
The PR https://github.com/grafana/loki/pull/721 doesn't fix it. It should fixes a different thing: modules initialization order determinism.

Reply:
Yes and that's all I need. When this is merged, the error message will always be the second one mentioned in the #721. This one is human readable and the actual error. 

Together with this one the whole topic is resolved.

Reply:
> [...] the error message will always be the second one mentioned in the #721.

I don't think so. `table-manager` should be initialized before the `store` module, so it will always panic instead of panic "most of the times".

This PR https://github.com/cortexproject/cortex/pull/1492 should fixes it.

Reply:
Ah! I should have taken a look at the code instead of just the issues 😅 

Then we shall wait for both, #721 and cortexproject/cortex#1492!